compiler/passes/src/code_generation/program.rs

```diff
@@ -362,7 +366,11 @@ impl<'a> CodeGeneratingVisitor<'a> {
         writeln!(mapping_string, "    value as {};", create_type(&mapping.value_type)).expect(EXPECT_STR);
 
         // Add the mapping to the variable mapping.
-        self.global_mapping.insert(mapping.identifier.name, mapping.identifier.to_string());
+        self.global_mapping.insert(
+            mapping.identifier.name,
+            Self::legalize_path(&[mapping.identifier.name])
```

Comment:
You could probably legalize once at the top. Your call on whether you'd find that improvement helpful.

Reply:
Done